src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -75,6 +77,13 @@ public void onPost() {
      * Called when a POST is made.
      */
     public void onPost(String triggeredByUser) {
+        onPost(SCMEvent.originOf(Stapler.getCurrentRequest()), triggeredByUser);
```

Comment:
So why not call `Stapler.getCurrentRequest()` in `GitHubTrigger` implementation, it still the same thread?